test/unit/controllers/user/OAuthControllers.spec.ts

```diff
@@ -0,0 +1,34 @@
+import { Response, Request } from 'express';
+import OAuthServices from 'src/services/user/OAuthServices';
+import logger from '@tablerise/dynamic-logger';
+import OAuthControllers from 'src/controllers/user/OAuthControllers';
+import { HttpStatusCode } from 'src/support/helpers/HttpStatusCode';
+
+describe('Controllers :: User :: OAuthControllers', () => {
+    const OAuthServicesMock = new OAuthServices(logger);
+    const OAuthControllersMock = new OAuthControllers(OAuthServicesMock, logger);
+
+    const request = {} as Request;
+    const response = {} as Response;
+
+    describe('When a request is made to authenticate with google', () => {
+        beforeAll(() => {
+            response.status = jest.fn().mockReturnValue(response);
+            response.json = jest.fn().mockReturnValue({});
+
+            jest.spyOn(OAuthServicesMock, 'google').mockResolvedValue({}); // change to user's mock when done
```

Comment:
Esse comentário ainda precisa estar aqui?

Reply:
Sim, é um lembrete pra quando essa feature da entidade do usuário for implementada